src/components/NFTViewOptionList.vue

```diff
@@ -156,9 +158,24 @@ export default {
     handleClickViewContent() {
       this.$emit('view-content');
     },
-    handleClickViewContentURL(e, contentUrl) {
+    async handleClickViewContentURL(e, contentUrl) {
       const type = this.getContentUrlType(contentUrl);
-      this.$emit('view-content-url', e, parseNFTMetadataURL(contentUrl), type);
+      const url = parseNFTMetadataURL(contentUrl);
+      this.$emit('view-content-url', e, url, type);
+      if (type !== 'pdf') {
```

Comment:
Should pdf's `router.push` to reader page be refactored into here also? otherwise actions are all spreaded out into different components
Also consider keeping the `<a>`  with `href` and `download` and use `e.preventDefault()` for more link context and accessibility

Reply:
Updated